### PR TITLE
[RW-6667][risk=no] API changes to support access module expiration and renewal

### PR DESCRIPTION
**Pull request by dmohs**
Description:
This adds expiration information for compliance training and the data use agreement.

Response to /v1/profile is hardcoded as "All modules will never expire and are not currently expired":
```
{
  ...
  renewableAccessModules: {
    modules: [
      {
        moduleName: 'complianceTraining',
        expirationEpochMillis: null,
        hasExpired: false
      },
      {
        moduleName: 'dataUseAgreement',
        expirationEpochMillis: null,
        hasExpired: false
      }
    ],
    anyModuleHasExpired: false
  }
}
```

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/resources/workbench-api.yaml [hunk 1]
```diff
@@ -5534,6 +5534,33 @@ definitions:
...
         description: Timestamp when the user was bypassed for completing login.gov linking
           linking.
+      renewableAccessModules:
+        modules:
+          type: array
+          items:
+            "$ref": "#/definitions/RenewableAccessModuleStatus"
+        compliance:
+          anyModuleHasExpired:
+            type: boolean
```
Comment (dmohs) [inline api/src/main/resources/workbench-api.yaml:5544]: This does duplicate the information available by scanning all modules in the array, but we believe it is useful to include this summary for clients and worth the tradeoff.

**Review by jmthibault79 — APPROVED** [review]
LGTM.  You can merge #4951 (rebased onto this) to add ProfileMapper + ProfileService updates for compilation.

**Comment by dmohs** [comment]
@jmthibault79 Thank you for your PR, which I heavily cribbed. I don't think I can merge it because I removed one unnecessary level ("compliance"). If you're okay with that change, I'll merge.

**Review by jmthibault79 — APPROVED** [review]
Works for me.  Needs a `./gradlew spotlessApply` for tests to pass

**Comment by dmohs** [comment]
@jmthibault79, your gentle mentorship style is delightful and very much appreciated.

**Pull request merged**